jerry-main/main-unix-snapshot.c

```diff
@@ -0,0 +1,413 @@
+        }
+      }
+    }
+
+    buffer_p += code_size;
+  }
+  while (buffer_p < buffer_end_p);
+} /* scan_snapshot_functions */
+
+/**
+ * Update all literal offsets in a snapshot data.
+ */
+static void
+update_literal_offsets (uint8_t *buffer_p, /**< snapshot buffer start */
+                        const uint8_t *buffer_end_p, /**< snapshot buffer start */
+                        lit_mem_to_snapshot_id_map_entry_t *lit_map_p) /**< literal map */
+{
+  JERRY_ASSERT (buffer_end_p > buffer_p);
+
+  do
+  {
+    ecma_compiled_code_t *bytecode_p = (ecma_compiled_code_t *) buffer_p;
+    uint32_t code_size = ((uint32_t) bytecode_p->size) << JMEM_ALIGNMENT_LOG;
+
+    if (bytecode_p->status_flags & CBC_CODE_FLAGS_FUNCTION)
+    {
+      jmem_cpointer_t *literal_start_p;
+      uint32_t const_literal_end;
+
+      if (bytecode_p->status_flags & CBC_CODE_FLAGS_UINT16_ARGUMENTS)
+      {
+        literal_start_p = (jmem_cpointer_t *) (buffer_p + sizeof (cbc_uint16_arguments_t));
+
+        cbc_uint16_arguments_t *args_p = (cbc_uint16_arguments_t *) buffer_p;
+        const_literal_end = args_p->const_literal_end;
+      }
+      else
+      {
+        literal_start_p = (jmem_cpointer_t *) (buffer_p + sizeof (cbc_uint8_arguments_t));
+
+        cbc_uint8_arguments_t *args_p = (cbc_uint8_arguments_t *) buffer_p;
+        const_literal_end = args_p->const_literal_end;
+      }
+
+      for (uint32_t i = 0; i < const_literal_end; i++)
+      {
+        if (literal_start_p[i] != JMEM_CP_NULL)
+        {
+          lit_mem_to_snapshot_id_map_entry_t *current_p = lit_map_p;
+
+          while (current_p->literal_id != literal_start_p[i])
+          {
+            current_p++;
+          }
+
+          literal_start_p[i] = current_p->literal_offset;
+        }
+      }
+    }
+
+    buffer_p += code_size;
+  }
+  while (buffer_p < buffer_end_p);
+} /* update_literal_offsets */
+
+/**
+ * Print help.
+ */
+static void
+print_help (char *name) /**< program name */
+{
+  printf ("Usage: %s [COMMAND] [ARGUMENTS]\n"
+          "\n"
+          "Command list:\n"
+          "  merge [OUTPUT_FILE] [INPUT_FILEs]...\n"
+          "      Merge input files into output file.\n"
+          "  help\n"
+          "      Prints this help.\n"
+          "\n",
+          name);
+} /* print_help */
+
+/**
+ * Main function.
+ *
+ * @return error code (0 - no error)
+ */
+int
+main (int argc, /**< number of arguments */
+      char **argv) /**< argument list */
+{
+  if (argc <= 1 || (argc == 2 && (!strcmp ("help", argv[1]))))
+  {
+    print_help (argv[0]);
+    return JERRY_STANDALONE_EXIT_CODE_OK;
+  }
+
+  if (strcmp ("merge", argv[1]))
+  {
+    printf ("Unknown command: %s\n", argv[1]);
+    print_help (argv[0]);
+    return JERRY_STANDALONE_EXIT_CODE_OK;
+  }
+
+  if (argc < 5)
+  {
+    printf ("At least an output and two input files must be passed\n");
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
```

Comment:
This is clearly wrong. The format of the parameters:

First, the command part:

Command: must be merge or help (no -h and such)

Second the arguments of the command.

help can have a single argument, the type of a command, e.g. merge. Currently it does not process the arguments

merge has an output file, and then input file arguments

The output is not an option, so it has no -o flag. At least two input files must be specified.

Fortunately I realized we don't need to use argument parsing in this tool, so we can just keep the current code. The only important thing is that we need add an option with two arguments to main-unix.c, how can I do that?